orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v2schema/model/V2PipelineTemplate.java

```diff
@@ -0,0 +1,307 @@
+
+    public List<String> getScopes() {
+      return scopes;
+    }
+
+    public void setScopes(List<String> scopes) {
+      this.scopes = scopes;
+    }
+  }
+
+  public static class Variable implements NamedContent<Variable>, Cloneable {
+    private String name;
+    private String group;
+    private String description;
+    private String type;
+    private Object defaultValue;
+    private String example;
+    private boolean nullable;
+    private boolean merge = false;
+    private boolean remove = false;
+
+    @Override
+    public String getName() {
+      return name;
+    }
+
+    public void setName(String name) {
+      this.name = name;
+    }
+
+    public String getGroup() {
+      return Optional.ofNullable(group).orElse("General");
+    }
+
+    public void setGroup(String group) {
+      this.group = group;
+    }
+
+    public String getDescription() {
+      return description;
+    }
+
+    public void setDescription(String description) {
+      this.description = description;
+    }
+
+    public String getType() {
+      return Optional.ofNullable(type).orElse("object");
+    }
+
+    public void setType(String type) {
+      this.type = type;
+    }
+
+    public Object getDefaultValue() {
+      return defaultValue;
+    }
+
+    public void setDefaultValue(Object defaultValue) {
+      this.defaultValue = defaultValue;
+    }
+
+    public boolean hasDefaultValue() {
+      return defaultValue != null;
+    }
+
+    public String getExample() {
+      return example;
+    }
+
+    public void setExample(String example) {
+      this.example = example;
+    }
+
+    public boolean isNullable() {
+      return nullable;
+    }
+
+    public void setNullable(boolean nullable) {
+      this.nullable = nullable;
+    }
+
+    @Override
+    public boolean isMerge() {
+      return merge;
+    }
+
+    public void setMerge(boolean merge) {
+      this.merge = merge;
+    }
+
+    @Override
+    public boolean isRemove() {
+      return remove;
+    }
+
+    public void setRemove(boolean remove) {
+      this.remove = remove;
+    }
+
+    @Override
+    public Variable merge(Variable overlay) {
+      Variable v;
+      try {
+        v = (Variable) this.clone();
+      } catch (CloneNotSupportedException e) {
+        throw new RuntimeException("Could not clone Variable", e);
+      }
+      if (overlay.group != null) { v.group = overlay.group; }
+      if (overlay.description != null) { v.description = overlay.description; }
+      if (overlay.type != null) { v.type = overlay.type; }
+      if (overlay.defaultValue != null) { v.defaultValue = overlay.defaultValue; }
+      if (overlay.example != null) { v.example = overlay.example; }
+      return v;
+    }
+
+    @Override
+    protected Object clone() throws CloneNotSupportedException {
+      return super.clone();
+    }
+  }
+
+  public static class Configuration {
+    private Map<String, Object> concurrentExecutions;
+    private List<NamedHashMap> triggers;
```

Comment:
I thought we were trying to avoid having to maintain a separate model & update it each time the pipeline schema changes?